shared/src/commonMain/kotlin/org/hyperskill/app/step_completion/presentation/StepCompletionFeature.kt

```diff
@@ -9,26 +9,39 @@ interface StepCompletionFeature {
         fun createState(step: Step, stepRoute: StepRoute): State =
             State(
                 currentStep = step,
-                continueButtonAction = if (stepRoute is StepRoute.Learn) {
-                    ContinueButtonAction.FetchNextStepQuiz
+                startPracticingAction = when (stepRoute) {
+                    is StepRoute.Learn.TheoryOpenedFromPractice, is StepRoute.Repeat.Theory ->
+                        StartPracticingAction.NavigateToBack
+                    is StepRoute.Learn.Step, is StepRoute.LearnDaily,
+                    is StepRoute.Repeat.Practice, is StepRoute.StageImplement ->
+                        StartPracticingAction.FetchNextStepQuiz
```

Comment:
Let's update formatting:

```suggestion
                    is StepRoute.Learn.TheoryOpenedFromPractice, 
                    is StepRoute.Repeat.Theory ->
                        StartPracticingAction.NavigateToBack
                    is StepRoute.Learn.Step, 
                    is StepRoute.LearnDaily,
                    is StepRoute.Repeat.Practice, 
                    is StepRoute.StageImplement ->
                        StartPracticingAction.FetchNextStepQuiz
```